chia/server/node_discovery.py

```diff
@@ -485,8 +488,8 @@ async def start(self):
 
     async def close(self):
         await self._close_common()
-        self.self_advertise_task.cancel()
-        self.address_relay_task.cancel()
+        self.cancel_task_safe(self.self_advertise_task)
+        self.cancel_task_safe(self.address_relay_task)
```

Comment:
technically, we should probably cancel all the `self.pending_tasks` too here. and in my opinion we should also "join" or wait for all the tasks owned by this object too. but that's a more radical change to belong here